Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -3503,13 +3503,26 @@ private void UpdateFX(float elapsedClockSeconds)
                 // Variable is proportional to angular speed, value of 10 means 1 rotation/second.
                 // If wheel is not slipping then use normal wheel speed, this reduces oscillations in variable1 which causes issues with sounds.
 
-                if (SteamEngines[i].AttachedAxle.IsWheelSlip)
+                if (((Train.TrainType == Train.TRAINTYPE.PLAYER && !Train.Autopilot) || Train.TrainType == Train.TRAINTYPE.AI_PLAYERDRIVEN) && (Simulator.UseAdvancedAdhesion && !Simulator.Settings.SimpleControlPhysics))
```

Comment:
```suggestion
                if (AdvancedAdhesionModel)
```

Not sure if that works, but that would be a much simpler check

Reply:
I am not comfortable with this change as I can't see where AdvancedAdhesionModel is set/rest by the Autopilot function. Thus I suspect that it could be possibly set to true whilst in Autopilot, which would not be desirable if Autopilot doesn't use the Advanced Adhesion(?).

Reply:
The idea for that variable was that whenever advanced adhesion is being used, this is set to true. For AI controlled trains or user-selected adhesion, it is set to false. The motivation was precisely to avoid all the checks for !SimpleControls&Settings.AdvancedAdhesion&&!AITrain that happen multiple times in the code, and use that variable instead. However, I'm not sure about whether that variable is correctly updated at all times, so let's keep it as is, as you suggest.